rust/src/passwd.rs

```diff
@@ -363,6 +367,86 @@ impl PasswdKind {
     }
 }
 
+/// Due to a prior bug, the build system had some deployments with a world-readable
+/// shadow file.  This fixes a given deployment.
+#[context("Fixing shadow permissions")]
+pub(crate) fn fix_shadow_perms_in_root(root: &Dir) -> Result<bool> {
+    let zero_perms = Permissions::from_mode(0);
+    let mut changed = false;
+    for path in ["etc/shadow", "etc/shadow-", "etc/gshadow", "etc/gshadow-"] {
```

Comment:
@HuijingHei found that we are not fixing `/usr/etc/shadow` and `/usr/etc/gshadow`. I think this means we need another patch right @cgwalters ?

Reply:
Conceptually those files are part of the immutable base image defaults, so we "can't"¹ change them per instance.

They need to get fixed on the build server.

Note also, that because we don't have any hardcoded passwords in our base images, it doesn't actually *matter* if `/usr/etc/shadow` is world readable because there's nothing there.

¹ Of course, we could mask them or something but I don't think it matters, we can just explain the above

Reply:
I guess since `useradd` does not really work in layering yet is not an issue either? and if people are adding layered hardcoded credentials manually to shadow... they just need to fix it "downstream".